src/devdocs2zim/generator.py

```diff
@@ -0,0 +1,281 @@
+import argparse
+import os
+import re
+from collections import defaultdict
+
+from pydantic import BaseModel
+from zimscraperlib.constants import (  # pyright: ignore[reportMissingTypeStubs]
+    MAXIMUM_DESCRIPTION_METADATA_LENGTH,
+    MAXIMUM_LONG_DESCRIPTION_METADATA_LENGTH,
+    RECOMMENDED_MAX_TITLE_LENGTH,
+)
+
+from devdocs2zim.client import (
+    DevdocsClient,
+    DevdocsMetadata,
+)
+from devdocs2zim.constants import logger
+
+
+class InvalidFormatError(Exception):
+    """Raised when a user supplied template has an invalid parameter."""
+
+    pass
+
+
+class ZimConfig(BaseModel):
+    """Common configuration for building ZIM files."""
+
+    # Directory the ZIM will be written to.
+    output_folder: str
```

Comment:
It hitches me a bit that output_folder is part of ZIM config. Not a big concern, I can live with it, but for me it is more a config of the scraper as a whole, since it will not go inside that ZIM like all other flags will.

Reply:
pulled this out